### PR TITLE
Add the ability to cache search results

### PR DESCRIPTION
**Pull request by adempsey**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @adempsey has signed the CLA. Thanks for the pull request!

**Comment by adempsey** [comment]
@RicoYao

**Comment by RicoYao** [comment]
+2

File: BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h [hunk 1]
```diff
@@ -292,6 +292,10 @@
 
 #pragma mark - Search
 
+- (void)cacheSearchRequest:(BOXSearchRequest *)request
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h:295]: Should this match the params of BOXItemArrayCompletionBlock? (i.e. totalCount, range, etc.)

Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h:295]: The answer is NO. Those params can be looked up from 'request' which has offset and limit as params.

**Comment by RicoYao** [comment]
+2

**Pull request merged**